### ISSUE TITLE
Remove obsolete no_stack_check attribute and test

### ISSUE THREAD

**Issue by infinity0** [rust-lang/rust#34915]
Here is the start of the failure

```
failures:

---- [debuginfo-gdb] debuginfo-gdb/function-prologue-stepping-no-stack-check.rs stdout ----


executing armv7-unknown-linux-gnueabihf/stage2/bin/rustc /home/infinity0/rust/src/test/debuginfo/function-prologue-stepping-no-stack-check.rs -L armv7-unknown-linux-gnueabihf/test/debuginfo-gdb/ --target=armv7-unknown-linux-gnueabihf -L armv7-unknown-linux-gnueabihf/test/debuginfo-gdb/function-prologue-stepping-no-stack-check.stage2-armv7-unknown-linux-gnueabihf.debuginfo-gdb.libaux -C prefer-dynamic -o armv7-unknown-linux-gnueabihf/test/debuginfo-gdb/function-prologue-stepping-no-stack-check.stage2-armv7-unknown-linux-gnueabihf -C link-args=-Wl,-z,relro --cfg rtopt -C rpath -L armv7-unknown-linux-gnueabihf/rt -g
------stdout------------------------------

------stderr------------------------------
/home/infinity0/rust/src/test/debuginfo/function-prologue-stepping-no-stack-check.rs:253:1: 253:18 warning: unused attribute, #[warn(unused_attributes)] on by default
/home/infinity0/rust/src/test/debuginfo/function-prologue-stepping-no-stack-check.rs:253 #[no_stack_check]
                                                                                         ^~~~~~~~~~~~~~~~~
/home/infinity0/rust/src/test/debuginfo/function-prologue-stepping-no-stack-check.rs:270:1: 270:18 warning: unused attribute, #[warn(unused_attributes)] on by default
/home/infinity0/rust/src/test/debuginfo/function-prologue-stepping-no-stack-check.rs:270 #[no_stack_check]
                                                                                         ^~~~~~~~~~~~~~~~~
[..] many of these, basically one for every instance of no_stack_check in the file
```

After this basically everything is broken which you can see in more detail [here](https://gist.github.com/infinity0/8ce7179052e1f0ce0131737015068112). However, I see that `no_stack_check` isn't used in the code anywhere any more - there is only one `grep` result for `stack[-_]check` that excludes this test itself, and one of them says this:

```
src/libsyntax/feature_gate.rs-    // Not used any more, but we can't feature gate it
src/libsyntax/feature_gate.rs:    ("no_stack_check", Normal, Ungated),
```

So I wonder if this test should just be gotten rid of?

**Comment by nagisa** [comment]
The test probably should go.

**Comment by nagisa** [comment]
I think its also a time to deprecate the attribute by now. cc @rust-lang/compiler @rust-lang/lang

**Comment by michaelwoerister** [comment]
If the attribute doesn't have any effect anymore, the test case should be removed (while making sure that's it's still possible to break on a function, which might mean that we have to write another test case if that isn't covered anywhere else).

**Comment by infinity0** [comment]
fwiw, this test *does* pass on x86, x86-64 and even arm64

**Comment by pnkfelix** [comment]
It seems odd that the test is failing in this context, but not in the other targets that @infinity0 mentioned.

Nonetheless, the compiler team concluded we should:

 1. kill this test,
 2. kill this attribute,
 3. kill any compiler flag that exposing a similar thingie.

**Comment by nikomatsakis** [comment]
I'll happily mentor someone to do this.

**Comment by abhijeetbhagat** [comment]
I can work on this.

**Comment by abhijeetbhagat** [comment]
So this is as simple as just deleting `/src/test/debuginfo/function-prologue-stepping-regular.rs` and removing this:
` // Not used any more, but we can't feature gate it`
` ("no_stack_check", Normal, Ungated),`
from feature_gate.rs ?

**Comment by nagisa** [comment]
You’ll also need to remove [this flag](https://github.com/rust-lang/rust/blob/9316ae515e2f8f3f497fb4f1559910c1eef2433d/src/librustc/session/config.rs#L640). Getting the things you mentioned along with this flag deleted sounds correct.

**Comment by abhijeetbhagat** [comment]
I created three different pull requests because i did not know how to create a single pull request for multiple files. How can i do this in the future?

**Comment by nagisa** [comment]
> I created three different pull requests because i did not know how to create a single pull request for multiple files. How can i do this in the future?

Did you perhaps make the PRs using [github’s edit file feature](https://help.github.com/articles/editing-files-in-your-repository/)? I do not think that feature was ever meant for anything more than quickly changing a typo in a single file.

The proper workflow (which allows for multi-file patches as well) is to:

* Fetch the project to your device; 
* Change the files (as many of them as you want);
* Commit the changes to your local repository using any of the [git tools](https://git-scm.com/downloads) available for your platform. Github has [its own GUI](https://desktop.github.com/) application for Windows and Mac as well;
* Push your commit(s) to your github fork;
* Create a PR for your commit(s).

There a number of tutorials on how to get started with this workflow on the internet. In no particular order: [1], [2], [2.5], [3], [4], [5].

[1]: https://www.thinkful.com/learn/github-pull-request-tutorial/
[2]: https://help.github.com/articles/fork-a-repo/
[2.5]: https://help.github.com/articles/using-pull-requests/
[3]: https://github.com/CPAN-PRC/resources/wiki/My-first-Pull-Request
[4]: https://guides.github.com/activities/hello-world/
[5]: https://gist.github.com/Chaser324/ce0505fbed06b947d962

**Comment by pmatos** [comment]
What's still missing here?

**Comment by lfairy** [comment]
@pmatos AFAIK, the only thing left to do is have `-C no-stack-check` print a deprecation warning when used. There was a pull request for this at #35156, but it was closed due to inactivity.

**Comment by mgattozzi** [comment]
This issue should be closed now right? It mentions in the release notes of https://github.com/rust-lang/rust/pull/38966 that this has been fixed and it seems like #34915 was the one that closed the issue.

**Comment by michaelwoerister** [comment]
@mgattozzi Yes, looks like it. Thanks for the reminder!

**Comment by nagisa** [comment]
No. We still need to remove the flag.

**Comment by michaelwoerister** [comment]
Oh, I thought the issue was just about the test, sorry.

**Comment by danobi** [comment]
This issue was fixed in PR #37636. I believe this should be closed.

**Comment by nikomatsakis** [comment]
per @danobi's comment, closing
